Checkmarx/sast-to-ast-export
# Feature/ast 10757 flatten group hierarchy

By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../CODE-OF-CONDUCT.md). Please see the [contributing guidelines](../CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

Flattens group hierarchy as needed in order to properly set permissions in AST.

### References

https://checkmarx.atlassian.net/browse/AST-10757

### Testing

Unit tests added

Manual test:
1. Create export package
2. Import in AST
3. Assert IAM has flattened groups
4. Edit group leaf user in IAM, add default-roles, view-projects-if-in-group, view-results-if-in-group, and view-scans-if-in-group roles
5. Access to AST as user edited in previous step
6. Assert user can only see projects that belong to the group the user is in

### Checklist
